### PR TITLE
Update native-build-tools dependencies

### PR DESCRIPTION
**Pull request by olpaw**
This PR updates two dependencies so that we do not depend anymore on artifacts with known vulnerabilities.

### CODE REVIEW

File: build-logic/common-plugins/src/main/kotlin/org.graalvm.build.java.gradle.kts [hunk 1]
```diff
@@ -51,7 +51,7 @@ plugins {
 
 java {
     toolchain {
-        languageVersion.set(JavaLanguageVersion.of(8))
+        languageVersion.set(JavaLanguageVersion.of(11))
```
[the remark this answers is not in the document]
Comment (melix) [reply build-logic/common-plugins/src/main/kotlin/org.graalvm.build.java.gradle.kts:54]: I think we should mention this in the release notes. This means the plugins won't run anymore on a build tool (Maven or Gradle) which runs Java 8.

**Review by melix — CHANGES_REQUESTED** [review]

File: build-logic/common-plugins/src/main/kotlin/org.graalvm.build.java.gradle.kts (hunk 1 above)
[the remark this answers is not in the document]
Comment (melix) [reply build-logic/common-plugins/src/main/kotlin/org.graalvm.build.java.gradle.kts:54]: By release notes I mean in https://github.com/graalvm/native-build-tools/blob/master/docs/src/docs/asciidoc/index.adoc

[the remark this answers is not in the document]
Comment (olpaw) [reply build-logic/common-plugins/src/main/kotlin/org.graalvm.build.java.gradle.kts:54]: Are you suggesting to update
native-gradle-plugin/README.md
native-maven-plugin/README.md
to mention this change?

[the remark this answers is not in the document]
Comment (melix) [reply build-logic/common-plugins/src/main/kotlin/org.graalvm.build.java.gradle.kts:54]: Updating the `index.adoc` file should be enough, that's where we keep track of changes.

[the remark this answers is not in the document]
Comment (olpaw) [reply build-logic/common-plugins/src/main/kotlin/org.graalvm.build.java.gradle.kts:54]: Fixed in https://github.com/graalvm/native-build-tools/pull/497/commits/f97461cee59ddee939fcbe75eef72afc0a77be89

File: docs/src/docs/asciidoc/index.adoc [hunk 2]
```diff
@@ -22,7 +22,8 @@ If you are using alternative build systems, see <<alternative-build-systems.adoc
 === Release 0.9.26
 
 * Relax GraalVM version check for dev versions
-* Prepare plugins for release of *GraalVM for JDK 21*. They no longer deploy any experimental options.
+* Prepare plugins for release of _GraalVM for JDK 21_. They no longer deploy any experimental options.
+* This release drops support for Java 8. From now on Java 11 or later is required.
```
Comment (fniephaus) [inline docs/src/docs/asciidoc/index.adoc:26]: How about?
```suggestion
* Bump Java compliance of the plugins from Java 8+ to Java 11+.
```

**Review by melix — APPROVED** [review]

**Pull request merged**